massa-execution-worker/src/execution.rs

```diff
@@ -295,8 +307,9 @@ impl ExecutionState {
 
         // append generated events to the final event store
         exec_out.events.finalize();
-        self.final_events.extend(exec_out.events);
-        self.final_events.prune(self.config.max_final_events);
+        let final_events_count = exec_out.events.0.len();
+        self.final_events_cache
```

Comment:
note: this is synchronous, so the speed of save_events is critical as it might slow down the whole slot execution